### PR TITLE
finding frequency/heart rate + tests

### PR DESCRIPTION
**Pull request by anikamukherji**
@mlp6 @suyashkumar Do you mind taking a look at my heart rate function? I am not sure if I am calculating it correctly

### CODE REVIEW

**Comment by anikamukherji** [comment]
resolves #8

File: tools/hrm_tools.py [hunk 1]
```diff
@@ -53,11 +55,9 @@ def autocorr_freq(signal, fs):
...
-    corr = last_half[last_half.size//2:]
-    d = np.diff(last_half)
+    corr = ret[ret.size//2:]
+    d = np.diff(corr)
     start = find(d > 0)[0]
     peak = np.argmax(corr[start:]) + start
-    # sampling frequency
     freq = fs/peak
-    return freq
+    return int(freq)
```
Comment (mlp6) [inline tools/hrm_tools.py]: Is there a reason why you expect `freq` to be an `int`?

File: hrm/heart_rate_monitor.py [hunk 2]
```diff
@@ -183,7 +183,6 @@ def find_heart_rate(self):
         t = times[0]
         last_sample = t[-1]
         # sampling frequency
-        fs = t.size/last_sample
+        fs = round(t.size/last_sample)
```
Comment (mlp6) [inline hrm/heart_rate_monitor.py]: This works... could also consider taking the mean of the `diff` of `t`.  These should yield the same values as what you did, but the latter approach can be more flexible when you need to accommodate issues with uneven `dt` intervals.

File: tests/test_tools.py [hunk 3]
```diff
@@ -65,3 +65,19 @@ def test_return_column():
...
+        import numpy as np
+        from tools.hrm_tools import autocorr_freq
+    except ImportError as e:
+        print("Necessary import failed: {}".format(e))
+        return
+    test1 = np.array([0, 0, 1, 0, 0, 1, 0, 0, 1, 0])
+    freq1 = autocorr_freq(test1, len(test1))
+    test2 = np.array([0, 0.1, 0.2, 1, 0.05, 0.1, 1, 0, 0.12, 1, 0])
+    freq2 = autocorr_freq(test2, len(test2))
+    assert freq1 == 3
```
Comment (mlp6) [inline tests/test_tools.py:82]: The "frequency" that you are calculating here is a bit misleading, but I see what you're doing.  We can chat about this more tomorrow, or you can hopefully pick Suyash's brain today to get some additional insights!

File: tools/hrm_tools.py (hunk 1 above)
Comment (anikamukherji) [reply tools/hrm_tools.py]: I wasn't sure how else to test it without rounding it down! I could round it in the testing function...?

Comment (mlp6) [reply tools/hrm_tools.py]: can you not assert something to a floating point value?

Comment (anikamukherji) [reply tools/hrm_tools.py]: It isn't always 3.0, it is sometimes some irrational decimal value

**Pull request merged**